### ISSUE TITLE
Time of treatments

### ISSUE THREAD

**Issue by mvankessel-EMC** [darwin-eu-dev/TreatmentPatterns#201]
Taking inspiration from the treatment over time plots Ed made in a TreatmentPatterns study earlier this year I figured I would take a look into what it would take to add that as a feature.

## Plots
Originally treatments were represented as a bar plot counting the amount of treatments, per (combination)treatment. These bar plots would be split out per time ranges, in this case 0-30, 0-90, 0-365.
![image](https://github.com/darwin-eu-dev/TreatmentPatterns/assets/112690347/d84dbc20-2401-490c-8655-8121591c5bbd)
![image](https://github.com/darwin-eu-dev/TreatmentPatterns/assets/112690347/077c791c-5b8e-45b2-93ea-0dd6035784ac)
![image](https://github.com/darwin-eu-dev/TreatmentPatterns/assets/112690347/bd8a2033-2653-4fc6-b1c9-7a0f4aa6b3bf)

I personally think a cleaner way to communicate the durations would be to represent them as boxplots.
![image](https://github.com/darwin-eu-dev/TreatmentPatterns/assets/112690347/fffc8c3a-c0a4-4a85-aad8-93bc7554a189)

We could even do this for all the pathways found
![image](https://github.com/darwin-eu-dev/TreatmentPatterns/assets/112690347/a0c1ff79-9383-498f-8f4a-343099f47db2)

## Code
The `export()` function would need to export some form of duration of the treatments. For data safety, we could pre-compute the quantiles per treatment and per pathway, and export that. Which could then be plotted out in a new function.

For the ShinyApp this would become its own module.

@AniekMarkus @rossdwilliams @edward-burn @daniprietoalhambra please let me know what you think.

**Comment by AniekMarkus** [comment]
@mvankessel-EMC I like your boxplots and agree with you; I find this visualisation clearer than multiple barplots.

Another visualisation to consider (giving even more insight) is the 'survival analysis' plots I suggested earlier in our meetings/ issue #7. See example figure below (x-axis would be durationEra and y-axis would be % of treatments still ongoing at a given time).

![survival_plot](https://github.com/darwin-eu-dev/TreatmentPatterns/assets/60397369/80331ff0-82bf-42d1-aef3-2267169beaf0)

**Comment by mvankessel-EMC** [comment]
@AniekMarkus So something like this:
![image](https://github.com/darwin-eu-dev/TreatmentPatterns/assets/112690347/eb9d873f-e68f-4eb3-b822-50f40a7096a4)

This would also be pretty easy to export and share, as we can just export the `survfit` model itself.
```
Call: survfit(formula = Surv(time = durationEra) ~ path, data = x)

                                             n events median 0.95LCL 0.95UCL
path=Acetaminophen                         206    206   37.0      35      60
path=Acetaminophen-Amoxicillin+Clavulanate   6      6   75.0      35      NA
path=Acetaminophen-Aspirin                  14     14   90.0      35      NA
path=Acetaminophen+Amoxicillin               2      2   38.0      35      NA
path=Amoxicillin                             2      2   35.0      NA      NA
path=Amoxicillin+Clavulanate                48     48   84.0      70      98
path=Amoxicillin+Clavulanate-Acetaminophen   1      1   89.0      NA      NA
path=Amoxicillin+Clavulanate-Aspirin         2      2  140.0      91      NA
path=Aspirin                               211    211   60.0      35      60
path=Aspirin-Acetaminophen                  12     12   37.0      30      NA
path=Aspirin-Amoxicillin+Clavulanate         6      6   80.5      70      NA
path=Clavulanate                             3      3   98.0      49      NA
```

**Comment by daniprietoalhambra** [comment]
Can I suggest you discuss this with @catalamarti as I am aware he is incorporating treatment discontinuation/persistence using CohortSurvival in DrugUtilisation?

**Comment by catalamarti** [comment]
Yes, this has been implemented in DrugUtilisation (summariseTreatment and treatmentDiscontinuation)

https://dpa-pde-oxford.shinyapps.io/DrugUtilisation/

https://darwin-eu-dev.github.io/DrugUtilisation/reference/index.html

**Comment by edward-burn** [comment]
It would be good to discuss next week what research questions are best answered by treatment patterns vs drug utilisation.

@mvankessel-EMC I like what you started with which is designed to describe duration eras, as that is the output of treatment patterns by design. 

I think survival analysis (to talk about adherence to treatment) fits more naturally in drug utilisation (as treatment patterns is not really a study of failure processes, in the sense that people have multiple eras, go from single therapy to combination which would not normally be considered an outcome event, and so on)

**Comment by rossdwilliams** [comment]
Agreed, maybe we can discuss this on thursday and come up wiht a set of guidelines to go along with the standard catalogue about what should be done using which package

**Comment by mvankessel-EMC** [comment]
`TreatmentPatterns::plotEventDuration()` has been added showing boxplots of duration per (combination) event. Over the entire pathway, and per event line.
